addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1741,7 +1758,7 @@ const RangeUserValueWidget = UnitUserValueWidget.extend({
      * @private
      */
     _onInputChange(ev) {
-        this._value = ev.target.value;
+        this._value = this.isCustomRange ? this.customValues[ev.target.value] : ev.target.value;
```

Comment:
Slight inconsistency: here you set `this._value` to the custom string value but in `setValue`, `this._value` is still holding a number. You have to choose. Not sure but what I would do is keeping `this._value` to always be a number like a normal range. So you won't have to change `_onInputChange` but need to adapt `getValue` to return the string custom values from the numerical value.